programs/stake/src/stake_state.rs

```diff
@@ -1033,7 +1033,8 @@ fn validate_delegated_amount(
     // With the `stake_allow_zero_undelegated_amount` feature, stake accounts
     // may be initialized with a lower balance, so check the minimum in this
     // function, on delegation.
-    if feature_set.is_active(&stake_allow_zero_undelegated_amount::id())
+    if (feature_set.is_active(&stake_allow_zero_undelegated_amount::id())
```

Comment:
Unrelated to this PR but do you mind updating the comment to remove the part that says "With the `stake_allow_zero_undelegated_amount` feature"? We already know that the withdraw/delegate trick can lead to state accounts being delegated with zero stake.

Reply:
Like this?
```
    // Previously, `initialize` checked that the stake account balance met
    // the minimum delegation amount. Stake accounts may be initialized
    // with a lower balance, so check the minimum in this function, on delegation.

```

Reply:
Oh we can probably drop the first sentence too..

> Stake accounts may be initialized with a stake amount below the minimum delegation so check that the minimum is met before delegation.

Reply:
Done!